tests/BenchmarkDotNet.Tests/Exporters/XmlSerializerTests.cs

```diff
@@ -0,0 +1,245 @@
+        }
+
+        [Theory]
+        [InlineData(null, null, typeof(ArgumentException))]
+        [InlineData(nameof(MockSource.Items), null, typeof(ArgumentException))]
+        [InlineData(nameof(MockSource.Items), " ", typeof(ArgumentException))]
+        [InlineData(nameof(MockSource.Items), "", typeof(ArgumentException))]
+        [InlineData(null, "MockItem", typeof(ArgumentException))]
+        [InlineData(" ", "MockItem", typeof(ArgumentException))]
+        [InlineData("", "MockItem", typeof(ArgumentException))]
+        public void WithCollectionItemNameThrowsGivenInvalidArguments(string collectionName, string itemName, Type exception)
+        {
+            IXmlSerializer serializer = new XmlSerializer(typeof(MockSource));
+            Assert.Throws(exception, () => serializer.WithCollectionItemName(collectionName, itemName));
+        }
+
+        [Theory]
+        [InlineData(null, typeof(ArgumentException))]
+        [InlineData(" ", typeof(ArgumentException))]
+        [InlineData("", typeof(ArgumentException))]
+        public void WithExcludedPropertyThrowsGivenNameIsNullOrWhiteSpace(string name, Type exception)
+        {
+            IXmlSerializer serializer = new XmlSerializer(typeof(MockSource));
+            Assert.Throws(exception, () => serializer.WithExcludedProperty(name));
+        }
+
+        [Theory]
+        [MemberData(nameof(SerializeTestData))]
+        public void SerializeThrowsGivenNullArguments(IXmlWriter writer, object source, Type exception)
+        {
+            IXmlSerializer serializer = new XmlSerializer(typeof(MockSource));
+            Assert.Throws(exception, () => serializer.Serialize(writer, source));
+        }
+
+        public static IEnumerable<object[]> SerializeTestData { get; } =
+            new List<object[]>
+            {
+                new object[] { null, null, typeof(ArgumentNullException) },
+                new object[] { null, new MockSource(), typeof(ArgumentNullException) },
+                new object[] { new MockXmlWriter(), null, typeof(ArgumentNullException) }
+            };
+
+        [Fact]
+        public void WritesElementStringGivenSimpleCollectionItem()
+        {
+            IXmlWriter writer = new MockXmlWriter();
+            var serializer = new XmlSerializer(typeof(SimpleItemSource));
+
+            serializer.Serialize(writer, new SimpleItemSource());
+            string actual = writer.ToString();
+
+            Assert.Contains("Item s1 Item", actual);
+            Assert.Contains("Item s2 Item", actual);
+        }
+
+        [Fact]
+        public void DoesNotWriteUnwriteableCollection()
+        {
+            IXmlWriter writer = new MockXmlWriter();
+            IXmlSerializer serializer = new XmlSerializer(typeof(UnwriteableCollectionSource));
+
+            serializer.Serialize(writer, new UnwriteableCollectionSource());
+            string actual = writer.ToString();
+
+            Assert.DoesNotContain(actual, "Items");
+        }
```

Comment:
a lot of great tests! We might even have better xml tests than the `corefx` repo now ;) 👍